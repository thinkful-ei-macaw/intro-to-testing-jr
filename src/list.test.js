import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';


//Snapshot Test
it('renders without crashing', () => {
  const div = document.createElement('div');
  
 
  ReactDOM.render(<List header = 'something'
  cards = {[{ id: 'b', title: 'Second card', content: 'lorem ipsum' },
  { id: 'c', title: 'Third card', content: 'lorem ipsum' }]} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

//Snapshot test
it('renders the UI as expected', () => {
  const tree = renderer
    .create(<List header = 'First String' cards = {[]}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });


