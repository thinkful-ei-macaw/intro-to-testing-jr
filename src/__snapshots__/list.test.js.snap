// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    Third Card
  </h3>
  <p>
    lorem ipsum
  </p>
</div>
`;

exports[`renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      First String
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
